pkg/route/ingress/ingress.go

```diff
@@ -688,24 +705,28 @@ func routeMatchesIngress(
 		route.OwnerReferences[0].APIVersion == "networking.k8s.io/v1"
 
 	if !match {
-		return false
+		return false, incompleteIngressToRouteRules
 	}
 
 	targetPort, err := targetPortForService(ingress.Namespace, path.Backend.Service, serviceLister)
 	if err != nil {
 		// not valid
-		return false
+		incompleteIngressToRouteRules = append(incompleteIngressToRouteRules, fmt.Sprintf("No valid target port for backend service %q", path.Backend.Service.Name))
+		return false, incompleteIngressToRouteRules
```

Comment:
I just noticed that we cannot have more than 1 element in this slice. I think we can simply return an `error` instead of  `[]string` and use it upper the callstack:
```suggestion
		return false, fmt.Errorf("No valid target port for backend service %q", path.Backend.Service.Name)
```